engine/access/rpc/engine.go

```diff
@@ -128,11 +130,20 @@ func New(log zerolog.Logger,
 	// wrap the unsecured server with an HTTP proxy server to serve HTTP clients
 	httpServer := NewHTTPServer(unsecureGrpcServer, config.HTTPListenAddr)
 
+	cacheSize := config.ConnectionPoolSize
+	if cacheSize == 0 {
+		cacheSize = 100
+	}
+	cache, err := lru.New(cacheSize)
+	if err != nil {
+		// fmt.Errorf("cannot create conenction pool cache %w", err)
```

Comment:
Uncertain about what the best way of handling failure to init the cache would be, also uncertain about a good default size to set the cache to (above)

Reply:
if thew Cache fails to initialize, so should the server, right? not being able to connect to EN/CN in it's current state should be fatal

Reply:
also let's start with a size like `10` and go from there